prog/vm/nexus.rb

```diff
@@ -395,43 +395,46 @@ def before_run
       bud Prog::Vnet::UpdateFirewallRules, {subject_id: vm.id}, :update_firewall_rules
       hop_wait_firewall_rules_before_run
     when "NotStarted", "Failed"
-      topo = vm.cloud_hypervisor_cpu_topology
-
-      # we don't write secrets to params_json, because it
-      # shouldn't be stored in the host for security reasons.
-      params_json = JSON.pretty_generate({
-        "vm_name" => vm_name,
-        "public_ipv6" => vm.ephemeral_net6.to_s,
-        "public_ipv4" => vm.ip4.to_s || "",
-        "local_ipv4" => local_ipv4,
-        "unix_user" => vm.unix_user,
-        "ssh_public_key" => vm.public_key,
-        "nics" => vm.nics.map { |nic| [nic.private_ipv6.to_s, nic.private_ipv4.to_s, nic.ubid_to_tap_name, nic.mac] },
-        "boot_image" => vm.boot_image,
-        "max_vcpus" => topo.max_vcpus,
-        "cpu_topology" => topo.to_s,
-        "mem_gib" => vm.mem_gib,
-        "ndp_needed" => host.ndp_needed,
-        "storage_volumes" => storage_volumes,
-        "swap_size_bytes" => frame["swap_size_bytes"]
-      })
-
       secrets_json = JSON.generate({
         storage: storage_secrets
       })
 
       # Enable KVM access for VM user.
       host.sshable.cmd("sudo usermod -a -G kvm #{q_vm}")
 
-      # put prep.json
-      host.sshable.cmd("sudo -u #{q_vm} tee #{params_path.shellescape}", stdin: params_json)
+      write_params_json
```

Comment:
For learning purpose, why do we write to a file instead of previous way or passing along with the secrets via stdin? You might want to write a sentence about it to the commit message.

Reply:
Also, where do we delete this file?